hub/api/dataset.py

```diff
@@ -98,40 +90,59 @@ def __getitem__(self, item: Union[str, int, slice, Index]):
         else:
             raise InvalidKeyTypeError(item)
 
-    def __setitem__(self, item: Union[slice, str], value):
-        if isinstance(item, str):
-            tensor_key = item
-
-            if tensor_exists(tensor_key, self.storage):
-                raise TensorAlreadyExistsError(tensor_key)
-
-            if isinstance(value, np.ndarray):
-                tensor_meta = tensor_meta_from_array(value, batched=True)
-                ds_meta = read_dataset_meta(self.storage)
-                ds_meta["tensors"].append(tensor_key)
-                write_dataset_meta(self.storage, ds_meta)
-                tensor = Tensor(tensor_key, self.storage, tensor_meta=tensor_meta)
-                self.tensors[tensor_key] = tensor
-                tensor.append(value, batched=True)
-                return tensor
-            else:
-                raise UnsupportedTensorTypeError(item)
-        else:
-            raise InvalidKeyTypeError(item)
+    def create_tensor(
+        self,
+        name: str,
+        htype: Optional[str] = None,
+        chunk_size: Optional[int] = None,
+        dtype: Optional[str] = None,
+        extra_meta: Optional[dict] = None,
+    ):
+        """Creates a new tensor in a dataset.
+
+                Args:
+                    name (str): The name of the tensor to be created.
+                    htype (str, optional): The type of the data for the tensor.
```

Comment:
"Type of the data" tells me `dtype`. Maybe we can mention explicitly this is **not** dtype and is an entirely separate concept?

something like

```suggestion
                    htype (str, optional): The preset meta values for a tensor are determined by it's `htype` (different from `dtype`). These presets may be overwritten by the `extra_meta` argument. If no `htype` is provided, the global tensor defaults are used and is not recommended.
```

Reply:
```suggestion
                    htype (str, optional): The class of data for the tensor. 
                        The defaults for other parameters are determined in terms of this value. 
                        For example, `htype="image"` would have `dtype` default to `uint8`.
                        These defaults can be overridden by explicitly passing any of the other parameters to this function.
```